pcu/PCU.h

```diff
@@ -1,14 +1,34 @@
 #ifndef SCOREC_PCU_H
 #define SCOREC_PCU_H
 
+#if defined(SCOREC_NO_MPI)
+#include "pcu_pnompi_types.h"
+// Remove MPI calls.
+#define MPI_Init(argc, argv) do { \
+(void) argc; \
+(void) argv; \
+} while (0)
+#define MPI_Finalize(void) 
+#ifdef __cplusplus
+extern "C"
```

Comment:
What is the scope of these macros?  We still want to let users of PUMI run their application in parallel even if PUMI is only built for serial execution.

I think we should add a test that builds an MPI hello world app and links against a 'SCOREC_NO_MPI=on' build, and loads a serial pumi mesh etc..  This will help ensure that the symbols don't leak in the long run.

Reply:
This is a really good point. Right now, those macros would spread to every source that includes PCU.h. I think a simple `#ifndef MPI_COMM_WORLD` may suffice in these cases. Then PUMI tests will use the stub MPI_Init and end-users can `#include <mpi.h>` before PCU.h and they will get the real versions.

I will work on making a test case so that we can observe the negative first before trying that fix.

Also, it would require a major notice to end-users (since include order is not usually important). What do you think?

Reply:
If building with SCOREC_NO_MPI=off then I assume that everything that worked prior to this PR would continue to work.  Is that correct?  If so, then I think we could count this as a new 'feature' with specific requirements (i.e., include order, no mpi headers, etc.).

Reply:
Also, I see the Developer Guide on the Wiki which I will update after merging this PR. Is the user guide LaTeX anywhere? A quick Ctrl+F doesn't show any reference to any of the `SCOREC_*` build options.

Reply:
> Should this be a GitHub action, similar to building the doc/ folder?

Yes.

> latex 

The latex files for the user guide are here:

https://github.com/SCOREC/pumi-docs/tree/master/userGuide